source/server/options_impl.cc

```diff
@@ -12,8 +12,22 @@
 #include "spdlog/spdlog.h"
 #include "tclap/CmdLine.h"
 
+// Can be overridden at compile time
+#ifndef ENVOY_DEFAULT_MAX_STATS
+#define ENVOY_DEFAULT_MAX_STATS 16384
+#endif
+
+// Can be overridden at compile time
+#ifndef ENVOY_DEFAULT_MAX_STAT_NAME_LENGTH
+#define ENVOY_DEFAULT_MAX_STAT_NAME_LENGTH 127
+#endif
+
+#if ENVOY_DEFAULT_MAX_STAT_NAME_LENGTH < 127
```

Comment:
Don't feel that strongly about it but not sure if we need this check here, since it will fail in the options check below.

Reply:
I thought it would be annoying if you can compile a build where the default value is an error.  I can take this back out though if you want.

Reply:
Nah I'm ambivalent. That's fine.